src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1036,43 +1033,43 @@ private TokenValidationResult ValidateToken(string token, JsonWebToken outerToke
                    && (tokenValidationResult.Exception as SecurityTokenUnableToValidateException).ValidationFailure != ValidationFailure.InvalidLifetime)
                    || tokenValidationResult.Exception.GetType().Equals(typeof(SecurityTokenSignatureKeyNotFoundException)))
                 {
-                    if (validationParametersCopy.ConfigurationManager.UseLastKnownGoodConfiguration
-                        && validationParametersCopy.ConfigurationManager.LastKnownGoodConfiguration != null
-                        && !(validationParametersCopy.ConfigurationManager.LastKnownGoodConfiguration == validationParametersCopy.Configuration))
+                    if (validationParameters.ConfigurationManager.UseLastKnownGoodConfiguration
+                        && validationParameters.ConfigurationManager.LastKnownGoodConfiguration != null
+                        && !(validationParameters.ConfigurationManager.LastKnownGoodConfiguration == currentConfiguration))
```

Comment:
>&& !(validationParameters.ConfigurationManager.LastKnownGoodConfiguration == currentConfiguration) [](http://example.com/codeflow?start=24&length=98)

Replace with validationParameters.ConfigurationManager.LastKnownGoodConfiguration != currentConfiguration

Reply:
should we use reference equals.

Reply:
@dannybtsai Done!

As far as as reference equals goes, the reasoning behind using it was to minimize computation time. Checking equality through the use of an Equals method would be more time consuming, but I can see wanting to do that in order to avoid cases where we obtain a new configuration but it's not any different than the previous. That might actually result in us attempting to validate more than necessary.

How likely is this to happen with the value we have automatically set for the automatic refresh interval?